File: Easy/HexToDecimal.java
import java.io.*;
public class Main {
    public static String solve(String input) {
        Integer result = 0;
        Integer weight = 1;

        for(Integer i = input.length() - 1; i >= 0; i--) {
            Integer digit;
            if(input.charAt(i) > '9') {
                digit = input.charAt(i) - 'a' + 10;
            }
            else {
                digit = input.charAt(i) - '0';
            }
            result += digit * weight;
            weight *= 16;
        }

        return result.toString();
    }
    
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            System.out.println(solve(line));
        }
    }
}

File: Easy/HexToDecimal.cpp
#include <iostream>
#include <fstream>
#include <sstream>

using namespace std;

string solve(const string &input) {
    int result = 0;
    int weight = 1;

    for(int i = input.length() - 1; i >= 0; i--) {
        char c = input[i];
        int digit = (c > '9') ? c - 'a' + 10 : c - '0';
        result += digit * weight;
        weight *= 16;
    }

    return to_string(result);
}

int main(int argc, const char * argv[]) {
    ifstream input_stream(argv[1]);
    string input;
    
    while(getline(input_stream, input)) {
        cout << solve(input) << endl;
    }
    
    return 0;
}
